templates/_helpers.yaml

```diff
@@ -164,6 +164,18 @@ proxy_cookie_domain                     off;
 proxy_redirect                          off;
 {{ end }}
 
+{{/*
+Create the name of the dag-server service account to use
+*/}}
+{{- define "astro.dagDeploy.serviceAccountName" -}}
+{{- if .Values.dagDeploy.serviceAccount.create -}}
+    {{ default (printf "%s-dag-server" (include "airflow.fullname" .)) .Values.dagDeploy.serviceAccount.name }}
```

Comment:
Seems like we should have one more test assertion for each of these conditions. Currently we only have two assertions about `docs[0]["spec"]["template"]["spec"]["serviceAccountName"]`, but there are three possible code paths here.

Reply:
added missing checks as well